File: python-app/presentation/factories/output_presenter_factory.py
from presentation.interfaces.i_output_presenter_factory import IOutputPresenterFactory
from presentation.formatters.output_formatter import OutputFormatter
from presentation.presenters.output_presenter import OutputPresenter
from presentation.presenters.i_output_presenter import IOutputPresenter
from presentation.streams.output_stream import OutputStream

class OutputPresenterFactory(IOutputPresenterFactory):
    def create_presenter(self) -> IOutputPresenter:
        formatter = OutputFormatter()
        stream = OutputStream()
        return OutputPresenter(formatter=formatter, stream=stream)

File: angular-app/src/core/presentation/factories/output-presenter-factory.ts
import { OutputFormatter } from '../formatters/output-formatter';
import { IOutputPresenterFactory } from '../interfaces/i-output-presenter-factory';
import { OutputPresenter } from '../presenters/output-presenter';
import { OutputStream } from '../streams/output-stream';

export class OutputPresenterFactory implements IOutputPresenterFactory {
  createPresenter(): OutputPresenter {
    const formatter = new OutputFormatter();
    const stream = new OutputStream();
    return new OutputPresenter(formatter, stream);
  }
}
